kernel/sql-federation/core/src/main/java/org/apache/shardingsphere/sqlfederation/compiler/converter/segment/expression/impl/TypeCastExpressionConverter.java

```diff
@@ -43,7 +43,25 @@ public Optional<SqlNode> convert(final TypeCastExpression segment) {
         if (!expression.isPresent()) {
             return Optional.empty();
         }
-        SqlTypeNameSpec sqlTypeName = new SqlBasicTypeNameSpec(SqlTypeName.valueOf(segment.getDataType().toUpperCase()), SqlParserPos.ZERO);
+        String typeName = segment.getDataType().toUpperCase();
+        switch (typeName) {
+            case "INT2":
```

Comment:
Why do this type convert?

Reply:
> Why do this type convert?

You can view the supported SQL data types in org/apache/calcite/sql/type/SqlTypeName.java. This file contains the enumeration of all the standard SQL data types that are supported by Calcite. If you want to use a non-standard SQL data type in your queries, you will need to map it to its closest standard SQL data type before passing it to Calcite.

Reply:
> > Why do this type convert?
> 
> You can view the supported SQL data types in org/apache/calcite/sql/type/SqlTypeName.java. This file contains the enumeration of all the standard SQL data types that are supported by Calcite. If you want to use a non-standard SQL data type in your queries, you will need to map it to its closest standard SQL data type before passing it to Calcite.

Ok, I got it. Is this conversion logic only valid for PostgreSQL? We should probably design the SPI by databaseType to isolate these logics.

Reply:
> > > Why do this type convert?
> > 
> > 
> > You can view the supported SQL data types in org/apache/calcite/sql/type/SqlTypeName.java. This file contains the enumeration of all the standard SQL data types that are supported by Calcite. If you want to use a non-standard SQL data type in your queries, you will need to map it to its closest standard SQL data type before passing it to Calcite.
> 
> Ok, I got it. Is this conversion logic only valid for PostgreSQL? We should probably design the SPI by databaseType to isolate these logics.

Not really, the same conversion issue may arise when using OpenGauss with Calcite.

Reply:
openGauss is a derivative version of PostgreSQL database, these types do not exist in sqlserver, oracle, so they are not common logic.